src/cascadia/TerminalSettingsModel/TerminalSettings.h

```diff
@@ -0,0 +1,221 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- TerminalSettings.h
+
+Abstract:
+- The implementation of the TerminalSettings winrt class. Provides both
+        terminal control settings and terminal core settings.
+Author(s):
+- Mike Griese - March 2019
+
+--*/
+#pragma once
+
+#include "TerminalSettings.g.h"
+#include "IInheritable.h"
+#include "../inc/cppwinrt_utils.h"
+#include <DefaultSettings.h>
+#include <conattrs.hpp>
+
+// fwdecl unittest classes
+namespace TerminalAppLocalTests
+{
+    class SettingsTests;
+}
+
+namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+{
+    struct TerminalSettings : TerminalSettingsT<TerminalSettings>, IInheritable<TerminalSettings>
+    {
+        TerminalSettings() = default;
+        TerminalSettings(const Model::CascadiaSettings& appSettings,
+                         guid profileGuid,
+                         const TerminalControl::IKeyBindings& keybindings);
+
+        static Model::TerminalSettings BuildSettings(const Model::CascadiaSettings& appSettings,
+                                                     const Model::NewTerminalArgs& newTerminalArgs,
+                                                     const TerminalControl::IKeyBindings& keybindings);
+
+        Model::TerminalSettings MakeChild() { return *CreateChild(); }
+        void SetParent(Model::TerminalSettings parent)
+        {
+            ClearParents();
+            com_ptr<TerminalSettings> parentImpl;
+            parentImpl.attach(get_self<TerminalSettings>(parent));
+            InsertParent(0, parentImpl);
+        }

```

Comment:
These are basically just exposing functions from `IInheritable`, except exposing them as projected types. Unfortunately, that's why the name of these has to be something similar, but not the same thing.

I could update `IInheritable` to return projected types, then expose these functions in the IDL. But it feels really weird to expose inheritance outside of TSM. I guess `TerminalSettings` gets to be special in that way?

Reply:
`TerminalSettings` should always have only one parent. So `SetParent` makes sense as a name, even though the contents look a bit weird (because IInheritable is designed to handle multiple parents)